component/async/kafka/simple/simple.go

```diff
@@ -93,16 +106,22 @@ func (f *Factory) Create() (async.Consumer, error) {
 		c.partitions = c.partitionsSinceDuration
 	}
 
+	if c.config.NotificationChannel != nil {
+		c.notif = c.config.NotificationChannel
+	}
+
 	return c, nil
 }
 
 // consumer members can be injected or overwritten with the usage of OptionFunc arguments.
 type consumer struct {
-	topic      string
-	cnl        context.CancelFunc
-	ms         sarama.Consumer
-	config     kafka.ConsumerConfig
-	partitions func(context.Context) ([]sarama.PartitionConsumer, error)
+	topic         string
+	cnl           context.CancelFunc
+	ms            sarama.Consumer
+	config        kafka.ConsumerConfig
+	partitions    func(context.Context) ([]sarama.PartitionConsumer, error)
+	notif         chan<- struct{}
+	latestOffsets []int64
```

Comment:
I could not follow the code very well to understand this, so let me ask you: is there a possibility that `latestOffsets` is accessed unsafely my multiple goroutines? If we have a test covering this then the data race detector should find it out for us.

Reply:
No, until [that point](https://github.com/teivah/patron/blob/461535773a4b6717054375ea8663f42f0c65ae74/component/async/kafka/simple/simple.go#L170), there's only one goroutine, then we pass a copy for each offset.